pkg/gadgets/run/tracer/tracer.go

```diff
@@ -401,13 +411,80 @@ func (t *Tracer) processEventFunc(gadgetCtx gadgets.GadgetContext) func(data []b
 			}
 			timestampsOffsets = append(timestampsOffsets, member.Offset.Bytes())
 		}
+
+		btfSpec, err := btf.LoadKernelSpec()
+		if err != nil {
+			logger.Warnf("Kernel BTF information not available. Enums won't be resolved to strings")
+		}
+
+		if enum, ok := member.Type.(*btf.Enum); ok {
+			if btfSpec != nil {
+				kernelEnum := &btf.Enum{}
+				if err = btfSpec.TypeByName(enum.Name, &kernelEnum); err == nil {
+					// Use kernel enum if found
+					enum = kernelEnum
+				}
+			}
+
+			var getter func(data []byte) uint64
+			typ := simpleTypeFromBTF(member.Type)
+			switch typ.Kind {
+			case types.KindUint8:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[uint8](data, member.Offset.Bytes()))
+				}
+			case types.KindUint16:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[uint16](data, member.Offset.Bytes()))
+				}
+			case types.KindUint32:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[uint32](data, member.Offset.Bytes()))
+				}
+			case types.KindUint64:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[uint64](data, member.Offset.Bytes()))
+				}
+			case types.KindInt8:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[int8](data, member.Offset.Bytes()))
+				}
+			case types.KindInt16:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[int16](data, member.Offset.Bytes()))
+				}
+			case types.KindInt32:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[int32](data, member.Offset.Bytes()))
+				}
+			case types.KindInt64:
+				getter = func(data []byte) uint64 {
+					return uint64(getAsInteger[int64](data, member.Offset.Bytes()))
+				}
+			}
+
+			fieldSetter := types.GetSetter[string](t.eventFactory, member.Name)
+			enumSetter := func(ev *types.Event, data []byte) {
+				val := getter(data)
```

Comment:
What happens if `getter` is `nil`?

Reply:
Same as in the other comment, the switch statement is handling all possible values, so `getter` won't never be nil.